pyanaconda/modules/subscription/runtime.py

```diff
@@ -570,3 +570,118 @@ def run(self):
+            # machine readable organization id
+            org_data.organization_id = single_org.get("id","")
+            # human readable organization name
+            org_data.name = single_org.get("displayName","")
+            # content access mode, where:
+            # - "entitlement" means regular mode requiring entitlements to be attached
+            # - "org_environment" means Simple Content Access mode
+            simple_content_access_enabled = False
+            content_access_mode = single_org.get("contentAccessMode","")
+            if content_access_mode == ParseOrganizationDataTask.SCA_ENABLED_STRING:
+                # running in Simple Content Access mode
+                org_data.simple_content_access_enabled = True
+            elif content_access_mode == ParseOrganizationDataTask.SCA_NOT_ENABLED_STRING:
+                # "classic" non-SCA content access mode requiring entitlements to be attached
+                org_data.simple_content_access_enabled = False
+            else:
+                # unknown content access mode, log a warning and default to non-SCA
+                log.warning("subscription: unknown content access mode {} detected "
+                            "for organization {}/{}, falling back to non SCA mode".format(
+                                content_access_mode,
+                                org_data.name,
+                                org_data.organization_id
+                            )
+                )
+                org_data.simple_content_access_enabled = False
+            # finally, append to the list
+            org_data_list.append(org_data)
+
+        return org_data_list
+
+    def run(self):
+        """Parse organization data for a Red Hat account username and password.
+
+        :raises: RegistrationError if calling the RHSM DBus API returns an error
+        """
+        log.debug("subscription: getting data about organizations")
+        with RHSMPrivateBus(self._rhsm_register_server_proxy) as private_bus:
+            locale = os.environ.get("LANG", "")
+            private_register_proxy = private_bus.get_proxy(RHSM.service_name,
+                                                           RHSM_REGISTER.object_path)
+            org_data_json = private_register_proxy.GetOrgs(self._username,
+                                                           self._password,
+                                                           {},
+                                                           locale)
+            log.debug("subscription: got organization data")
+
+            org_data_length = 0
+            # Log how much organization data we got for debugging purposes.
+            # By only logging length, we should be able to debug cases of no
+            # or incomplete data being logged, without logging potentially
+            # sensitive subscription status detail into the installation logs
+            # stored on the target system.
```

Comment:
Is this really necessary?

Reply:
We already have this for the other calls that work with JSON input and it seems to me safer to have it in place so we can do basic debugging based on log output if needed (eq. zero length JSON, very long JSON (possibly getting a HTML error page instead of the JSON document we expect, etc.).

Reply:
In that case, could you please remove these eleven lines of code and comments and extend the debugging message above? You can do simply:

`log.debug("subscription: got organization data (%d characters)", len(org_data_json))`

It really doesn't deserve so much attention and creates the illusion, that there is something important going on.